EPRN.PRNS.API/Repositories/Repository.cs

```diff
@@ -278,6 +282,36 @@ public async Task<DraftDetailsPrnDto> GetDraftDetails(int id)
                 .SingleOrDefaultAsync();
         }
 
+        public async Task<List<PrnDto>> GetDraftPrnDetailsForUser(string userReferenceId)
+        {
+            var myPrns = new List<PrnDto>();    
+
+            var prns = await _prnContext
+                .PRN
+                .Include(p => p.WasteType)
+                .Where(prn => prn.UserReferenceId == userReferenceId)
+                .Select(prn => new PrnDto
+                {
+                    Id = prn.Id,
+                    PrnNumber = prn.Reference,
+                    Material = prn.WasteType.Name,
+                    SentTo = prn.SentTo,
+                    DateCreated = prn.CreatedDate.ToString("dd/MM/yyyy"),
+                    Tonnes = prn.Tonnes,
+                    Status = Common.Enums.PrnStatus.Draft
+                })
+                .ToListAsync();
+
+            foreach (var prn in prns)
+            {
+                var history = await _prnContext.PRNHistory.Where(x => x.PrnId == prn.Id).OrderByDescending(x => x.Created).FirstOrDefaultAsync();
```

Comment:
This is going to create an SQL query for each prn in the list, which could be very inefficient. There's probably a better way of doing this